packages/flutter/lib/src/widgets/spell_check.dart

```diff
@@ -99,80 +100,106 @@ class SpellCheckConfiguration {
+      final SuggestionSpan adjustedSpan = SuggestionSpan(
+          TextRange(start: adjustedSpanStart, end: adjustedSpanEnd),
+          currentSpan.suggestions,
       );
+
+      // Start search for the next misspelled word at the end of the
+      // adjusted currentSpan.
+      searchStart = adjustedSpanEnd + 1;
+      // Adjust offset to reflect the difference between where currentSpan
+      // was positioned in resultsText versus in newText.
+      offset = adjustedSpanStart - currentSpan.range.start;
       correctedSpellCheckResults.add(adjustedSpan);
     } else {
-      // Search for currentSpan in new text and if found, apply it to new text
-      // by adding it to the list of corrected results.
-      regex = RegExp('\\b$currentSpanText\\b');
-      foundIndex = newText.substring(searchStart).indexOf(regex);
-
-      if (foundIndex >= 0) {
-        foundIndex += searchStart;
-        spanLength = currentSpan.range.end - currentSpan.range.start;
-        searchStart = foundIndex + spanLength;
-        adjustedSpan = SuggestionSpan(
-            TextRange(start: foundIndex, end: searchStart),
-            currentSpan.suggestions
-        );
-        offset = foundIndex - currentSpan.range.start;
+      // Check if word was modified by extension by searching for word
+      // anywhere in the remaing substring of newText left to search.
+      final RegExp modifiedRegex = RegExp(currentSpanText);
+      final int modifiedFoundIndex = newText.substring(searchStart).indexOf(modifiedRegex);
+
+      if (modifiedFoundIndex >= 0) {
+        // Word was modified by extension, so check for next space following the
+        // modified word to find its end index.
+        final RegExp regex = RegExp(' ');
+        final int adjustedSearchStartForSpace = modifiedFoundIndex + searchStart + spanLength;
+        final int foundEndIndex = newText.substring(adjustedSearchStartForSpace).indexOf(regex);
+        
+        if (foundEndIndex >= 0) {
+          // Word was found within newText.
+          final int adjustedSpanStart = modifiedFoundIndex + searchStart;
+          final int adjustedSpanEnd = adjustedSearchStartForSpace + foundEndIndex;
+          final SuggestionSpan adjustedSpan = SuggestionSpan(
+            TextRange(start: adjustedSpanStart, end: adjustedSpanEnd),
+            currentSpan.suggestions,
+          );
+
+          // Start search for the next misspelled word at the end of the
+          // adjusted currentSpan.
+          searchStart = adjustedSpanEnd + 1;
+          correctedSpellCheckResults.add(adjustedSpan);
+          // Adjust offset to reflect the difference between where currentSpan
+          // was positioned in resultsText versus where the modified version
+          // is positioned in newText.
+          offset = (adjustedSpanEnd - adjustedSpanStart) - (currentSpan.range.end - currentSpan.range.start) + (adjustedSpanStart - currentSpan.range.start);
```

Comment:
The one thing I still want to think about a little more is whether or not to calculate offset this way. The rest of the method assumes really only that one modification has been made between `resultsText` and `newText`, and while it may account for more cases than that, this explicitly allows for two operations: modification and movement.

Reply:
Thinking about this more, this algorithm covers the following cases:

 1. One or more `SuggestionSpan` shifts, e.g. "Hlello, wrold" --> "Hlello, beautiful wrold" or --> "Oh! Hlello, big, beautiful wrold"
 2. One or more `SuggestionSpan` modifications, e.g. "Hello, wrold" --> "Hello, wroldd" or --> "Helllo, wroldd"
 3. One or more `SuggestionSpan` shifts or modifications, e.g. "Hello, wrold" --> "Hello, big wroldd"

This should be all of the cases that we need to consider because we only care about finding the `SuggestionSpan` text coming from the old text (`resultsText`) in the new text, which means that we only care about shifts and modifications of that `SuggestionSpan` text. Calculating the offset this way in this case, means that we catch case 2 and 3. Case 1 is caught above in the code by adjusting `offset` based on the actual `foundIndex` of the `SuggestionSpan` text. Thus, the code does not at all assume that only one modification has been made, and I believe it is fine as is. I still think multiple modifications/shifts is unlikely, but it doesn't hurt to cover that case.

Reply:
How are you able to do cases 2 and 3?  If I'm understanding correctly, those show that a misspelled word has been modified (`wrold` to `wroldd`), and you're still able to correct it (`world`)?  That seems super tricky if I'm understanding it correctly.  It might be better to not handle those at all.  What if the modification creates a correctly spelled word (`worl` to `worlds`)?

Reply:
Not quite -- case 2 means that a word like `wrold` has been modified to `wroldd`, and it would correct the 
`SuggestionSpan` of that word from `SuggestionSpan(TextRange(n, n+5), <String>["world"])` to  `SuggestionSpan(TextRange(n, n+6), <String>["world"])`, for example. It's this case (and 3 because 3 involves 2) that pushed me to change this method because if you are typing faster than the spell check results update, the red underline flashes under the misspelled word because of the lag (the cursor index/base offset would not be considered as touching the misspelled word you are composing because an "unknown" addition to the word has been made). My changes fix this issue in the case you described. You would still be compoinsg `worl` --> `worlds`, but none of it would be underlined in red, unless after composing, it is still misspelled.

Reply:
Ok I think I got it, we definitely do want the ability to tell that the word is still being edited in these cases so we can hide the red underline.